subcommand/server-acl-init/command_test.go

```diff
@@ -721,318 +638,100 @@ func TestRun_BindingRuleUpdates(t *testing.T) {
-	require.NoError(err)
-	port, err := strconv.Atoi(serverURL.Port())
-	require.NoError(err)
+	bootToken := getBootToken(t, k8s, resourcePrefix, ns)
 
-	// The pods and statefulset are created but as an in-progress deployment
-	pods := k8s.CoreV1().Pods(ns)
-	_, err = pods.Create(&v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: resourcePrefix + "-server-0",
-			Labels: map[string]string{
-				"component": "server",
-				"app":       "consul",
-				"release":   releaseName,
-			},
-		},
-		Status: v1.PodStatus{
-			PodIP: serverURL.Hostname(),
-		},
-		Spec: v1.PodSpec{
-			Containers: []v1.Container{
-				{
-					Name: "consul",
-					Ports: []v1.ContainerPort{
-						{
-							Name:          "http",
-							ContainerPort: int32(port),
-						},
-					},
-				},
-			},
-		},
+	// Check that it has the right policies.
+	consul, err := api.NewClient(&api.Config{
+		Address: srv.HTTPAddr,
 	})
 	require.NoError(err)
-	_, err = k8s.AppsV1().StatefulSets(ns).Create(&appv1.StatefulSet{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: resourcePrefix + "-server",
-			Labels: map[string]string{
-				"component": "server",
-				"app":       "consul",
-				"release":   releaseName,
-			},
-		},
-		Status: appv1.StatefulSetStatus{
-			UpdateRevision:  "updated",
-			CurrentRevision: "current",
-		},
-	})
+	tokenData, _, err := consul.ACL().TokenReadSelf(&api.QueryOptions{Token: bootToken})
 	require.NoError(err)
+	require.Equal("global-management", tokenData.Policies[0].Name)
 
-	ui := cli.NewMockUi()
-	cmd := Command{
-		UI:        ui,
-		clientset: k8s,
-	}
-	cmd.init()
-
-	// Start the command before the Pod exist.
-	// Run in a goroutine so we can create the Pods asynchronously
-	done := make(chan bool)
-	var responseCode int
-	go func() {
-		responseCode = cmd.Run([]string{
-			"-server-label-selector=component=server,app=consul,release=" + releaseName,
-			"-resource-prefix=" + resourcePrefix,
-			"-k8s-namespace=" + ns,
-			"-expected-replicas=1",
-		})
-		close(done)
-	}()
-
-	// Asynchronously update the deployment status after a delay.
-	go func() {
-		// Update after a delay between 100 and 500ms.
-		// It's randomized to ensure we're not relying on specific timing.
-		delay := 100 + rand.Intn(400)
-		time.Sleep(time.Duration(delay) * time.Millisecond)
-		_, err = k8s.AppsV1().StatefulSets(ns).Update(&appv1.StatefulSet{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: resourcePrefix + "-server",
-				Labels: map[string]string{
-					"component": "server",
-					"app":       "consul",
-					"release":   releaseName,
-				},
-			},
-			Status: appv1.StatefulSetStatus{
-				UpdateRevision:  "updated",
-				CurrentRevision: "updated",
-			},
-		})
-		require.NoError(err)
-	}()
-
-	// Wait for the command to exit.
-	select {
-	case <-done:
-		require.Equal(0, responseCode, ui.ErrorWriter.String())
-	case <-time.After(2 * time.Second):
-		require.FailNow("command did not exit after 2s")
+	// Check that the agent policy was created.
```

Comment:
might be able to use the `policyExists` helper

Reply:
Nice! Good catch!